src/main/java/com/google/sps/servlets/AddOrganizationServlet.java

```diff
@@ -52,15 +72,23 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     openHours.add(Integer.parseInt(hourOpen));
     openHours.add(Integer.parseInt(hourClosed));
 
-    ArrayList<String> moderatorList = new ArrayList<String>();
-    moderatorList.add("anon creator"); //dummy value for before auth gets set up
+    /* This implementation stores history entries as embedded entities instead of custom objects
+     * because it is much simpler that way */
+    ArrayList changeHistory = new ArrayList<>();
+    
+    EmbeddedEntity historyEntry = new EmbeddedEntity();
```

Comment:
May I suggest extracting a function that you can use later:

class HistoryManager {
  void PrepareForOrganization(Entity organization);
  void RecordHistory(userId, message, timestamp);
}

My imagined use case would be:

// This will read the current history and set up its own state if one exists. It also calls
// newOrganization.setProperty("changeHistory", ...); to point to local memory
HistoryManager history = new HistoryManager(newOrganization);

// This creates the embedded entity and adds it to the history
history.RecordHistory(userId, "Org was registered", millisecondsSinceEpoch);

newOrganization.setProperty(...); // all other changes

That way you can use the history manager later as well.

Reply:
I wrote this class to just have the function recordHistory, which takes in the parameters and then returns an entity that can be added to the history arraylist from wherever it is being called.